test/unit/specs/data/query-builder.js

```diff
@@ -264,6 +264,12 @@ describe('<Data> Query Builder', function() {
     expect(() => query.setOffset(-1)).to.throw(errorMsg)
   })
 
+  it('fails when distinct is invalid', function() {
+    const errorMsg = 'Distinct must be a boolean value.'
+
+    expect(() => query.setDistinct('foobar')).to.throw(errorMsg)
```

Comment:
what about other types?

Reply:
When I try to invoke API calls with `distinct = 'foobar'` that works the same as `false`. So I think this error makes sense.

Reply:
no, I meant why don't you check this method with other types such as `number` `array` etc?